src/MessagePack.UnityClient/Assets/Scripts/MessagePack/MessagePackSerializer.cs

```diff
@@ -462,58 +467,171 @@ private static int LZ4Operation(in ReadOnlySequence<byte> input, Span<byte> outp
+                        const int LengthOfUncompressedDataSizeHeader = 5;
+                        writer.WriteExtensionFormatHeader(new ExtensionHeader(ThisLibraryExtensionTypeCodes.LZ4, LengthOfUncompressedDataSizeHeader + (uint)lz4Length));
+                        writer.WriteInt32((int)msgpackUncompressedData.Length);
+                        writer.WriteRaw(lz4Span.AsSpan(0, lz4Length));
+                    }
+                    finally
+                    {
+                        ArrayPool<byte>.Shared.Return(lz4Span);
+                    }
                 }
-                finally
+                else if (compression == MessagePackCompression.Lz4ContiguousBlock)
                 {
-                    ArrayPool<byte>.Shared.Return(lz4Span);
+                    // Write to [Ext(98:int,int...), bin,bin,bin...]
+                    // Reading count from Sequence<T> is better but not yet do.
+                    var sequenceCount = 0;
+                    foreach (var item in msgpackUncompressedData)
+                    {
+                        sequenceCount++;
+                    }
+
+                    writer.WriteArrayHeader(sequenceCount + 1);
+                    writer.WriteExtensionFormatHeader(new ExtensionHeader(ThisLibraryExtensionTypeCodes.LZ4Contiguous, sequenceCount * 5));
+                    {
+                        foreach (var item in msgpackUncompressedData)
+                        {
+                            // force write Int32 block(size = 5).
+                            writer.WriteInt32(item.Length);
+                        }
+                    }
+
+                    foreach (var item in msgpackUncompressedData)
+                    {
+                        var maxCompressedLength = LZ4Codec.MaximumOutputLength(item.Length);
+                        var lz4Span = ArrayPool<byte>.Shared.Rent(maxCompressedLength);
+                        try
+                        {
+                            int lz4Length = LZ4Codec.Encode(item.Span, lz4Span);
+                            writer.Write(lz4Span.AsSpan(0, lz4Length));
+                        }
+                        finally
+                        {
+                            ArrayPool<byte>.Shared.Return(lz4Span);
+                        }
+                    }
+                }
+                else
+                {
+                    throw new MessagePackSerializationException("Invalid MessagePackCompression Code.");
                 }
             }
         }
+
+        private static class PrimitiveChecker<T>
+        {
+            public static readonly bool IsFixedSizePrimitive;
+
+            static PrimitiveChecker()
+            {
+                IsFixedSizePrimitive = IsFixedSizePrimitiveTypeHelper(typeof(T));
+            }
+        }
+
+        private static bool IsFixedSizePrimitiveTypeHelper(Type type)
+        {
+            return type == typeof(short)
```

Comment:
Should `Guid`, `decimal` be part of this set? If not, can we add xml doc comments to the API to explain how you define "primitive"?

Reply:
Primitive means MsgPack primitive.
I think we can add it (DateTime etc.)
I wanted to avoid types that providing custom formatter in builtins.